### PR TITLE
[ls-qpack] Add new port port

### PR DESCRIPTION
**Pull request by talregev**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by talregev** [comment]
Windows build is fail on my local. It not create the lib file, and I am not sure why.

**Comment by dg0yt** [comment]
> Windows build is fail on my local. It compile without errors, but it not create the lib file, and I am not sure why.

If there is no `__declspec(dllexport)` and no `.def` file and no  `WINDOWS_EXPORT_ALL_SYMBOLS`, then MSVC will not produce an import lib for the DLL.

**Comment by talregev** [comment]
> > Windows build is fail on my local. It compile without errors, but it not create the lib file, and I am not sure why.
> 
> If there is no `__declspec(dllexport)` and no `.def` file and no `WINDOWS_EXPORT_ALL_SYMBOLS`, then MSVC will not produce an import lib for the DLL.

Then it mean I should create `.def` file as I saw in other ports.

**Comment by talregev** [comment]
> WINDOWS_EXPORT_ALL_SYMBOLS

Adding 
`-DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON`

solved the problem! thank you! :)

We should also apply that on nettle port. and not edit manually the `.def` file.

File: ports/ls-qpack/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,28 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO litespeedtech/ls-qpack
+    REF v${VERSION}
```
Comment (WangWeiLin-MV) [inline ports/ls-qpack/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

File: ports/ls-qpack/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+vcpkg_find_acquire_program(PKGCONFIG)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        "-DPKG_CONFIG_EXECUTABLE=${PKGCONFIG}"
+        -DLSQPACK_TESTS=OFF
+        -DLSQPACK_BIN=OFF
+        -DLSQPACK_XXH=OFF
+        -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON
```
Comment (WangWeiLin-MV) [inline ports/ls-qpack/portfile.cmake:20]: > Libraries that don't provide a .def file and do not use __declspec() declarations simply do not support shared builds for Windows and should be marked as such with vcpkg_check_linkage(ONLY_STATIC_LIBRARY).

https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-add-cmake_windows_export_all_symbols

**Comment by dg0yt** [comment]
> > WINDOWS_EXPORT_ALL_SYMBOLS
> 
> Adding `-DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON`
> 
> solved the problem! thank you! :)

No! I made a different proposal.

**Comment by talregev** [comment]
> > > WINDOWS_EXPORT_ALL_SYMBOLS
> > 
> > 
> > Adding `-DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON`
> > solved the problem! thank you! :)
> 
> No! I made a different proposal.

I didn't understand that. I will change your suggestion. Thank you!

File: ports/ls-qpack/portfile.cmake (hunk 1 above)
Comment (talregev) [reply ports/ls-qpack/portfile.cmake]: done.

File: ports/ls-qpack/portfile.cmake (hunk 2 above)
Comment (talregev) [reply ports/ls-qpack/portfile.cmake]: done.

File: ports/ls-qpack/portfile.cmake
Comment (WangWeiLin-MV) [inline ports/ls-qpack/portfile.cmake:1]: Since the upstream has not released the export file yet, please provide usage file, such as `find_library`.

Comment (talregev) [reply ports/ls-qpack/portfile.cmake:1]: ok. I will add find_library, I hope it will find it.

Comment (talregev) [reply ports/ls-qpack/portfile.cmake:1]: Done.

**Comment by talregev** [comment]
@WangWeiLin-MV Ready for review.

File: ports/ls-qpack/usage [hunk 3]
```diff
@@ -0,0 +1,3 @@
+
+    find_library(LSQPACK ls-qpack)
+    target_link_libraries(main PRIVATE ${LSQPACK})
```
Comment (WangWeiLin-MV) [inline ports/ls-qpack/usage:3]:
```suggestion
The package libdatrie can be used via CMake:

    find_path(LSQPACK_INCLUDE_DIRS "lsqpack.h")
    find_library(LSQPACK_LIBRARY ls-qpack REQUIRED)
    
    target_include_directories(main PRIVATE ${LSQPACK_INCLUDE_DIRS})
    target_link_libraries(main PRIVATE ${LSQPACK_LIBRARY})
```

Comment (talregev) [reply ports/ls-qpack/usage:3]: Thank you!

Comment (talregev) [reply ports/ls-qpack/usage:3]: Done.

File: ports/ls-qpack/usage [hunk 4]
```diff
@@ -0,0 +1,6 @@
+The package ls-qpack can be used via CMake:
+    find_path(LSQPACK_INCLUDE_DIRS "lsqpack.h")
+    find_library(LSQPACK_LIBRARY ls-qpack REQUIRED)
+
```
Comment (WangWeiLin-MV) [inline ports/ls-qpack/usage:4]:
```suggestion
The package ls-qpack can be used via CMake:

    find_path(LSQPACK_INCLUDE_DIRS "lsqpack.h")
    find_library(LSQPACK_LIBRARY ls-qpack REQUIRED)

```

Comment (WangWeiLin-MV) [reply ports/ls-qpack/usage:4]: Extra spaces removed.

Comment (talregev) [reply ports/ls-qpack/usage]: I added a new line as you suggested. I hope now it ok.
Let me know.

**Review by WangWeiLin-MV — APPROVED** [review]
The port usage tests pass with the following triplets:

* x64-linux
* x64-windows

**Pull request merged**